Readability.js

```diff
@@ -1616,6 +1618,44 @@ Readability.prototype = {
     }
   },
 
+  /* convert images and figures that have properties like data-src into images that can be loaded without JS */
+  _fixLazyImages: function(root) {
+    var nodesToCheck = [].slice.call(root.getElementsByTagName("img"))
+    .concat([].slice.call(root.getElementsByTagName("picture")))
+    .concat([].slice.call(root.getElementsByTagName("figure")))
+
+    this._forEachNode(nodesToCheck, function(elem) {
+      if ((!elem.src && !elem.srcset) || elem.className.indexOf("lazy") !== -1) {
+        for (var i = 0; i < elem.attributes.length; i++) {
+          var attr = elem.attributes[i]
+          if(attr.name === "src" || attr.name === "srcset") {
+            continue;
+          }
+          var copyTo = null;
+          if(/.*\.(jpg|png|bmp|tiff)\s\d+/.test(attr.value)) {
+            copyTo = "srcset"
+          } else if (/https?:\/\/.*\.(jpg|png|bmp|tiff)/.test(attr.value)) {
+            copyTo = "src"
+          }
+          if(copyTo) {
+            //if this is an img or picture, set the attribute directly
+            if (elem.tagName === "IMG" || elem.tagName === "PICTURE") {
+              elem.setAttribute(copyTo, attr.value)
+            } else {
+              //if the item is a <figure> that does not contain an image or picture, create one and place it inside the figure
+              //see the nytimes-3 testcase for an example
+              if (elem.tagName === "FIGURE" && elem.getElementsByTagName("img").length === 0 && elem.getElementsByTagName("picture").length === 0) {
+                var img = this._doc.createElement("img")
+                img.setAttribute(copyTo, attr.value)
+                elem.appendChild(img)
```

Comment:
It would be better to make the image the first child in the figure, but that would require using `insertBefore`, which JSDOMParser doesn't support.

Reply:
Yes, the NYT markup has a caption and some other elements inside the figure, so the caption ends up being above the image. Adding insertBefore in a separate PR sounds fine to me.